### PR TITLE
Implemented dynamic module loading and hooks system for cmake

### PR DESCRIPTION
**Pull request by coolzoom**

### CODE REVIEW

**Comment by talamortis** [comment]
Are you able to test @coolzoom

**Comment by vhiperdev** [comment]
image

image

@talamortis @coolzoom Work!

**Comment by talamortis** [comment]
awesome, thanks for test we just need to try multiple modules to make sure everything is working as intended.

**Comment by coolzoom** [comment]
Probably does not able to support the one with config file yet, wonder do we keep conf separate one by one(will be a lot according to the mod number) or do we add them to core.conf(core conf will be super long)? Oh yes, seems no one is better, Guess just keep them separated. That’s the next step we can do.

**Comment by yuanf225** [comment]
cool,this will allow OC to be modularized as well

**Comment by talamortis** [comment]
Yes it will allow all C++ scripts to be dragged and drop without any core edits :)

**Comment by talamortis** [comment]
i am hoping to have this complete by the weekend, with DB driven config files

**Comment by talamortis** [comment]
@coolzoom  are you able to test and remove it so the module does not add / create the config files in the main folder?

**Comment by coolzoom** [comment]
> @coolzoom are you able to test and remove it so the module does not add / create the config files in the main folder?
working!
image
image

image

**Comment by talamortis** [comment]
Can you remove it from the cmake file?

**Comment by DanielBallaSZTE** [comment]
It would be nice if you followed the current OregonCore style with 2 spaces inserted for indentation.

**Comment by coolzoom** [comment]
> Can you remove it from the cmake file?

already removed, I reverted that commit

**Comment by talamortis** [comment]
Sorry daniel I prefer 4 spaces, otherwise it messes with my autism, plus it looks and reads alot easier

**Comment by DanielBallaSZTE** [comment]
I'm fine with 4 spaces either, just please change tab characters to spaces.

**Comment by coolzoom** [comment]
@talamortis can we auto-import the sql file inside modules folder?

**Comment by coolzoom** [comment]
@DanielBallaSZTE Daniel do you mean I have wrong indentations inside my commits?

File: src/game/World.cpp [hunk 1]
```diff
@@ -399,6 +399,33 @@ Weather* World::AddWeather(uint32 zone_id)
...
+				_moduleConfig[mod.config] = mod;
+
+				count++;
+			} while (result->NextRow());
+		}
+
+		sLog.outString(">> Loaded %lu module config", count);
+	}
+}
+
```
Comment (DanielBallaSZTE) [inline src/game/World.cpp]: Yes, right here for example you are using TAB characters instead of 4 spaces.

Comment (coolzoom) [reply src/game/World.cpp]: ok, i changed them https://github.com/talamortis/OregonCore/pull/123/commits/0e77e90f80394dd8ded03100eeff5bc3009004e1

**Comment by coolzoom** [comment]
@talamortis can you check the DatabaseUpdater for modules i added?

**Comment by talamortis** [comment]
Can this be merged now @coolzoom

**Comment by coolzoom** [comment]
> Can this be merged now @coolzoom

yes I think so, I tested on my computer and works, but would be nice if you can review the ace get directories function.

**Review by coolzoom — COMMENTED** [review]
looks to be fine

**Comment by talamortis** [comment]
Looks fine to me :D. Thanks Again for your support! Will merge :):)

**Pull request merged**